test/registerEsbuild.js

```diff
@@ -33,6 +33,12 @@ function compileFor(loader) {
       format: 'cjs',
       sourcefile,
       loader,
+      tsconfigRaw: {
```

Comment:
What's the added benefit for this? Do we need it for the new test?

Reply:
Yes because esbuild doesn't transpile class properties correctly, they use assign semantics instead of define, so the new test doesn't fail without this

Reply:
Detailed explanation:

```ts
// This vanilla JS syntax
class User {
  id;
  age = 28;
}

// Is equivalent to writing this

class User {
  constructor() { 
    Object.defineProperties(this, {
    id: { value: undefined },
    age: { value: 28 },
});
  }
}
```

This `defineProperty` semantic does not call setters if one exists on the prototype chain. Instead it just adds the properties on the object.

esbuild is not standard compliant by defaults and transpiled it to this:

```ts
class User {
  constructor() { 
    // Undefined property omitted
    this.age = 28;
  }
}
```

Properties with undefined as the value, which is how you'd declare a model attribute, are omitted by esbuild, preventing the test from failing.
Non-undefined values use assign semantics, which call the setter we added on the prototype of the model. It's not good, but users aren't supposed to set a value and it's only an issue with esbuild.

The configuration tweak I made tells esbuild to output spec compliant code

Reply:
Thanks!